src/plugins/auto_batch/tests/functional/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME ov_auto_batch_func_tests)
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    ie_add_compiler_flags(/wd4244)
+    ie_add_compiler_flags(/wd4267)
+    ie_add_compiler_flags(/wd4305)
+    ie_add_compiler_flags(/wd4018)
+    ie_add_compiler_flags(/wd4050)
+    ie_add_compiler_flags(/wd4250)
+    ie_add_compiler_flags(/wd4334)
+    ie_add_compiler_flags(/wd4661)
+    ie_add_compiler_flags(/wd4273)
+    ie_add_compiler_flags(/wd4309)
+    ie_add_compiler_flags(/wd4804)
+endif()
+if(NOT MSVC)
+    ie_add_compiler_flags(-Wno-missing-declarations)
+    ie_add_compiler_flags(-Wno-sign-compare)
+    ie_add_compiler_flags(-Wno-odr)
+    ie_add_compiler_flags(-Wno-all)
+endif()
+
+set(SHARED_HEADERS_DIR "${OpenVINO_SOURCE_DIR}/src/tests/functional/plugin/shared/include")
+addIeTargetTest(
+        NAME
+            ${TARGET_NAME}
+        ROOT
+            ${CMAKE_CURRENT_SOURCE_DIR}
+        INCLUDES
+            ${CMAKE_CURRENT_SOURCE_DIR}
+            ${SHARED_HEADERS_DIR}
+        LINK_LIBRARIES
+            funcSharedTests
+        DEPENDENCIES
+            openvino_auto_batch_plugin
+        ADD_CPPLINT
+        LABELS
+            Auto_Batch
+)
+
+if (ENABLE_INTEL_GPU)
+    find_package(OpenCL QUIET)
+    if(OpenCL_FOUND)
+        target_compile_definitions(${TARGET_NAME} PRIVATE HAVE_INTEL_GPU_PLUGIN)
```

Comment:
why do we need to check OpenCL?

Reply:
why do we need to split auto_batch tests from GPU or CPU?
I would treat having BATCH tests in cpuFUncTests are integration tests that CPU can work under auto-batch..

Reply:
1. Check OCL is to decide whether GPU plugin can be used, you can see that not all CI docker configure OCL which will cause GPU plugin cannot be loaded, and GPU plugin related auto_batch test cases failed.
2.  I'm not sure whether it is good idea to split auto_batch tests from GPU and CPU functional testcases, but if we will put all auto_batch related test cases into a separated test sample, it will be easier to maintain. Maybe we can still keep original auto_batch test cases in CPU/GPU functional test and add more auto_batch test cases into the separated test sample.

Reply:
AUTO_BATCH as a software feature as well its tests, should not not depend on GPU or OpenCL availability.
To check whether GPU plugin provides proper interface for communication with BATCH plugin, we need write tests specifically for this interface (e.g. whether optimal_batch_size is supported, maybe some other metrics)

Reply:
GPU or OCL dependents have been removed.
And we will add some tests for communication between AUTO_BATCH plugin and other plugins